gradio/processing_utils.py

```diff
@@ -325,10 +304,98 @@ def file_to_json(file_path):
     return json.load(open(file_path))
 
 
+class TempFileManager:
+    """
+    A class that should be inherited by any Component that needs to manage temporary files.
+    It should be instantiated in the __init__ method of the component.
+    """
+
+    def __init__(self) -> None:
+        # Set stores all the temporary files created by this component.
+        self.temp_files = set()
+
+    def hash_file(self, file_path: str, chunk_num_blocks: int = 128) -> str:
+        sha1 = hashlib.sha1()
+        with open(file_path, "rb") as f:
+            for chunk in iter(lambda: f.read(chunk_num_blocks * sha1.block_size), b""):
+                sha1.update(chunk)
+        return sha1.hexdigest()
+
+    def hash_url(self, url: str, chunk_num_blocks: int = 128) -> str:
+        sha1 = hashlib.sha1()
+        remote = urllib.request.urlopen(url)
+        max_file_size = 100 * 1024 * 1024  # 100MB
+        total_read = 0
+        while True:
+            data = remote.read(chunk_num_blocks * sha1.block_size)
+            total_read += chunk_num_blocks * sha1.block_size
+            if not data or total_read > max_file_size:
+                break
+            sha1.update(data)
+        return sha1.hexdigest()
+
+    def get_temp_file_path(self, file_path: str) -> str:
+        file_name = os.path.basename(file_path)
+        prefix, extension = file_name, None
+        if "." in file_name:
```

Comment:
For the bug bash, maybe we can refactor this to make use of pathlib `stem`, `root`, `suffix` properties of paths to not have to create this logic ourselves in multiple methods.

Reply:
Yes let me create an issue for this